countdown/module.py

```diff
@@ -0,0 +1,256 @@
+from datetime import datetime
+from typing import Optional, List
+
+import dateutil.parser
+import dateutil.relativedelta
+import discord
+from discord.ext import commands
+
+from pie.utils import ConfirmView
+from .database import CountdownItem
+from pie import check, i18n, logger, utils
+
+_ = i18n.Translator("modules/fun").translate
+bot_log = logger.Bot.logger()
+guild_log = logger.Guild.logger()
+
+
+class Countdown(commands.Cog):
+    """Countdown"""
+
+    def __init__(self, bot):
+        self.bot = bot
+
+    async def _process_text(self, ctx: commands.Context, text: Optional[str]):
+        if text is not None and len(text) > 1024:
+            text = text[:1024]
+            text = text[:-3] + "```" if text.count("```") % 2 != 0 else text
+
+        return text
+
+    async def _get_member(self, user_id: int, guild_id: int = 0):
+        user = None
+
+        if guild_id > 0:
+            guild = self.bot.get_guild(guild_id)
+            user = guild.get_member(user_id)
+
+        if user is None:
+            try:
+                user = await self.bot.fetch_user(user_id)
+            except discord.errors.NotFound:
+                pass
+
+        return user
+
+    def _get_remaining_time(self, countdown_item: CountdownItem) -> str:
+        if countdown_item.countdown_date < datetime.now():
+            return "Finished"
+        else:
+            return utils.time.format_seconds(
+                (countdown_item.countdown_date - datetime.now()).total_seconds()
+            )
+
+    async def _get_embed(self, ctx, countdown: CountdownItem):
+        embed = utils.discord.create_embed(
+            author=ctx.author,
+            title=_(ctx, "Countdown {name}").format(name=countdown.name),
+            description=countdown.message,
+        )
+        embed.add_field(
+            name=_(ctx, "Date"),
+            value=utils.time.format_datetime(countdown.countdown_date),
+        )
+        embed.add_field(
+            name=_(ctx, "Remaining time"), value=self._get_remaining_time(countdown)
+        )
+        embed.add_field(
+            name=_(ctx, "Created on"),
+            value=utils.time.format_datetime(countdown.origin_date),
+            inline=False,
+        )
+        embed.add_field(name=_(ctx, "URL"), value=countdown.permalink)
+
+        return embed
+
+    async def _send_countdown_list(self, ctx, query):
+        countdowns = []
+
+        for item in query:
+
+            countdown = CountdownDummy()
```

Comment:
You don't need to create new class, in Python you can just add attributes into new `countdown = object()`